### ISSUE TITLE
Support for recursive flag

### ISSUE THREAD

**Issue by rightaway** [so-fancy/diff-so-fancy#394]
If using `diff --recursive` only the first result has the proper formatting of the file names `Renamed: file1 to file2` with the coloring etc. Would be good to see it on all files and have full support of recursive flag, including when a file is in one directory and not the other.

**Comment by scottchiefbaker** [comment]
You're talking about vanilla diff? Not `git diff` output?

**Comment by rightaway** [comment]
@scottchiefbaker Yes for both vanilla diff and git diff, I'd like to pass it to diff-so-fancy even if it's `--recursive` and have formatted result.

**Comment by scottchiefbaker** [comment]
Apologies, this slipped through the cracks. If you can provide more information I might be able to tackle this. I'm not familiar with `diff --recursive`. How do you use it in your workflow? Can you provide a simple diff that show d-s-f providing the wrong output?

**Comment by scottchiefbaker** [comment]
Here is a possible simple test case (I think).

```diff
Only in /var/tmp/b: donk
diff --recursive -u /var/tmp/a/foo/bar /var/tmp/b/foo/bar
--- /var/tmp/a/foo/bar  2026-03-13 21:35:28.492228513 -0700
+++ /var/tmp/b/foo/bar  2026-03-13 21:35:47.101220189 -0700
@@ -1 +1 @@
-BAR
+FOO
diff --recursive -u /var/tmp/a/index.txt /var/tmp/b/index.txt
--- /var/tmp/a/index.txt        2026-03-13 21:35:20.997231861 -0700
+++ /var/tmp/b/index.txt        2026-03-13 21:35:41.030222906 -0700
@@ -1 +1,2 @@
 INDEX
+INDEX2
```

**Comment by scottchiefbaker** [comment]
This should be fixed in the latest v1.4.6 release. If not, please re-open.
